### ISSUE TITLE
Question about reproducing the reported results

### ISSUE THREAD

**Issue by pigeonmg** [zhoujun-7/VPHO#4]
Hi, thank you for sharing the code.

I tried to reproduce the reported results using the default settings and the training command provided in the README on DexYCB dataset. However, my reproduced performance is still much higher than the results reported in the paper.

My reproduced results are shown in the screenshot below.

Could you please let me know whether there are any additional details not mentioned in the README, such as specific hyperparameters, data preprocessing, or evaluation settings?

Thank you very much for your help.

<img width="240" height="140" alt="Image" src="https://github.com/user-attachments/assets/c31027c4-c1ea-4eaa-85e4-f1172f748109" />

<img width="689" height="656" alt="Image" src="https://github.com/user-attachments/assets/4c7fc5b0-cdc9-4b92-bf06-1a3d41770e40" />

**Comment by zhoujun-7** [comment]
Hi, thanks for reaching out and for your detailed report.

It seems your reproduced results are significantly higher than ours, which means the model may not have been trained properly under the exact setup used.

Could you share more details about your reproduction setup? For example:
- Training epoch
- Batch size
- OS version
- Hardware (GPU model, number of GPUs, etc.)

For reference, the results reported in the paper were obtained by training for 45 epochs with a batch size of 64×2, on a server running Ubuntu 22.04 with 2 NVIDIA RTX 4090 GPUs.

**Comment by zhoujun-7** [comment]
Additionally, could you try reproducing the results using our official checkpoint?

Note that the evaluation metrics logged during training are not calculated on the full dataset and use settings optimized for speed rather than accuracy.  

You will need to manually run the full evaluation script to obtain accurate, comparable results. You can launch the full evaluation with the following command:
```
accelerate launch --config_file lib/configs/single.yaml \
    main.py \
    --mode eval \
    --model vpho_net \
    --sample_T0 0.65 \
    --sample_num 100 \
    --topk_hand 30 \
    --topk_obj 10 \
    --batch_size 64 \
    --sampling_steps 50 \
    --checkpoint <path_to_checkpoint>/epoch_45.state
```

**Comment by pigeonmg** [comment]
Hi, thanks for your reply and clarification.

Here are my reproduction details:

* **OS:** Ubuntu 20.04.3
* **GPUs:** 2 × NVIDIA RTX 3090

I followed the training command in the README and kept the settings as close to the default configuration as possible. Some key parameters are:

```python
{   'RGB_shift_prob': 0.5,
    'aggregation_mode_hand': 'heatmap_cascade',
    'aggregation_mode_obj': 'heatmap_cascade',
    'base_learning_rate': 0.0002,
    'batch_size': 64,
    'bbox_scale_factor': 1.2,
    'blur_limit': (3, 7),
    'brightness': (0.6, 1.3),
    'center_jittering': 0.2,
    'checkpoint': '',
    'clahe_prob': 0.5,
    'clean_data_mode': '2023_CVPR_HFL',
    'color_jitter_prob': 0.5,
    'contact_normal_distance_thresh': (-0.01, 0.01),
    'contact_vertical_distance_thresh': 0.005,
    'contrast': (0.6, 1.3),
    'data_dir': './data/DexYCB',
    'dataset_name': 'dexycb',
    'do_physics_selection': True,
    'do_weighted_average': True,
    'eval_batch_size': 32,
    'eval_full': False,
    'eval_num_workers': 8,
    'eval_path': '',
    'eval_repeat_num': 50,
    'full_evaluation_freq': 10,
    'gamma': 0.96,
    'gaussian_blur_prob': 0.5,
    'gradient_accumulation_steps': 1,
    'gradient_clip': -1.0,
    'heatmap_hand_sigma': 2.0,
    'heatmap_obj_sigma': 2.0,
    'heatmap_size': 64,
    'hue': (-0.15, 0.15),
    'img_size': (640, 480),
    'lr_step': 5,
    'mark': 'vpho',
    'max_epochs': 100,
    'max_rot': 30,
    'mode': 'train',
    'model': 'vpho_net',
    'motion_blur_limit': (3, 7),
    'motion_blur_prob': 0.5,
    'num_workers': 16,
    'optimizer': 'adamw',
    'output_dir': 'output',
    'patch_size': 256,
    'pretrain': '',
    'print_freq': 500,
    'random_erasing_max_area': 0.2,
    'random_erasing_max_count': 1,
    'random_erasing_min_area': 0.02,
    'random_erasing_mode': 'pixel',
    'random_erasing_prob': 0.5,
    'random_seed': 206,
    'remove_pretrained_keys': [],
    'repeat_num': 20,
    'roi_size': 32,
    'rot_prob': 1,
    'sample_T0': 0.65,
    'sample_num': 50,
    'sampler': 'ode',
    'sampling_steps': 500,
    'saturation': (0.6, 1.3),
    'scale_factor': 0.2,
    'scheduler': 'exp',
    'sde_mode': 've',
    'shift_limit': (-20, 20),
    'sigma_limit': (0.2, 2.0),
    'start_with_eval': False,
    'topk_hand': 15,
    'topk_obj': 5,
    'use_mix_trainset': False,
    'use_regression_as_candidate': True,
    'viz_freq': 50,
    'weight_CoM_loss': 100.0,
    'weight_diff_hand_loss': 1.0,
    'weight_diff_obj_loss': 1.0,
    'weight_distrib_loss': 1.0,
    'weight_force_loss': 1.0,
    'weight_gravity_loss': 1.0,
    'weight_hand_contact_loss': 1.0,
    'weight_hm_hand_loss': 1000.0,
    'weight_hm_obj_loss': 1000.0,
    'weight_joint2hm_loss': 1000.0,
    'weight_joint_loss': 10000.0,
    'weight_mano_pose_loss': 10,
    'weight_mano_shape_loss': 1.0,
    'weight_obj_reg_kpt_loss': 10000.0,
    'weight_obj_reg_rot6d_loss': 10,
    'weight_obj_reg_trans_loss': 10000.0,
    'weight_obj_reg_vert_loss': 10000.0,
    'weight_point_loss': 100.0,
    'weight_segm_obj_loss': 30.0,
    'weight_supervised_loss': 10,
    'weight_torque_loss': 30.0,
    'weight_vert_loss': 10000.0}
```

I also tested the official checkpoint, and it can reproduce the reported results on my side.

The main issue is that when I train the model myself, I cannot obtain results comparable to the official checkpoint. Also, the training log metrics during my training are noticeably worse than those of the provided checkpoint, and the final evaluated performance is also worse.

So it seems there may still be some differences in the training setup.

Could you please let me know whether there are any additional training details not included in the README, such as preprocessing differences, checkpoint selection strategy, mixed precision settings, or other settings used for the paper results?

Thank you very much for your help.

**Comment by zhoujun-7** [comment]
Thanks for the information. I’ve uploaded the [training log file](https://github.com/zhoujun-7/VPHO/blob/main/training.log), which contains all the training settings for your reference. Note that this log was generated well before we drafted the manuscript, so some naming conventions may not be consistent with the current code version.

**Comment by pigeonmg** [comment]
Thanks for sharing the training log file.

I have compared the provided log with my reproduction setup, and the training parameters appear to be mostly consistent.

However, the intermediate evaluation results during my reproduction are still significantly worse than the corresponding intermediate evaluation results in the provided training log file.

This suggests that there may still be some differences affecting the training process, even though the parameter settings look largely the same.

For example, at epoch 45, the log from my reproduction is as follows:
```
2026-04-02 14:08:34,901 
Epoch: 45/100 | Learning Rate: 3.186e-05
2026-04-02 14:08:46,530 
[0000/3136]| Loss diff_H:4.10e+00 diff_O:3.99e-01 hm_H:1.33e+00 hm_O:9.59e-01 vert:1.50e-02 joint:1.50e-02 mano_pose:2.42e-03 mano_shape:1.09e-03 force:2.39e-02 gravity:1.38e-02 torque:1.98e-02 supervised:4.69e-02 CoM:6.35e-03 total:6.94e+00
2026-04-02 14:15:35,999 
[0500/3136]| Loss diff_H:3.51e+00 diff_O:3.49e-01 hm_H:1.19e+00 hm_O:8.36e-01 vert:1.36e-02 joint:1.42e-02 mano_pose:2.26e-03 mano_shape:1.65e-03 force:7.49e-02 gravity:6.05e-02 torque:3.39e-02 supervised:6.64e-02 CoM:5.16e-03 total:6.15e+00
2026-04-02 14:22:26,473 
[1000/3136]| Loss diff_H:3.70e+00 diff_O:4.25e-01 hm_H:1.20e+00 hm_O:8.98e-01 vert:1.48e-02 joint:1.43e-02 mano_pose:2.32e-03 mano_shape:6.83e-04 force:1.00e-01 gravity:7.22e-02 torque:2.31e-02 supervised:7.37e-02 CoM:4.61e-03 total:6.53e+00
2026-04-02 14:29:15,220 
[1500/3136]| Loss diff_H:4.21e+00 diff_O:4.42e-01 hm_H:1.40e+00 hm_O:1.03e+00 vert:1.41e-02 joint:1.41e-02 mano_pose:2.72e-03 mano_shape:1.48e-03 force:7.27e-02 gravity:4.76e-02 torque:3.08e-02 supervised:6.05e-02 CoM:4.90e-03 total:7.33e+00
2026-04-02 14:36:03,368 
[2000/3136]| Loss diff_H:3.63e+00 diff_O:4.35e-01 hm_H:1.38e+00 hm_O:9.79e-01 vert:1.46e-02 joint:1.58e-02 mano_pose:2.48e-03 mano_shape:2.97e-03 force:5.42e-02 gravity:3.31e-02 torque:2.90e-02 supervised:6.00e-02 CoM:5.33e-03 total:6.64e+00
2026-04-02 14:42:51,317 
[2500/3136]| Loss diff_H:3.64e+00 diff_O:4.29e-01 hm_H:1.22e+00 hm_O:8.83e-01 vert:1.32e-02 joint:1.37e-02 mano_pose:2.36e-03 mano_shape:9.69e-04 force:6.48e-02 gravity:5.08e-02 torque:3.44e-02 supervised:6.47e-02 CoM:4.43e-03 total:6.43e+00
2026-04-02 14:49:39,370 
[3000/3136]| Loss diff_H:3.54e+00 diff_O:4.06e-01 hm_H:1.21e+00 hm_O:9.19e-01 vert:1.41e-02 joint:1.48e-02 mano_pose:2.02e-03 mano_shape:1.17e-03 force:2.64e-02 gravity:1.18e-02 torque:3.04e-02 supervised:7.31e-02 CoM:5.51e-03 total:6.26e+00
2026-04-02 14:51:27,718 
Evaluation on subtest dataset: 7877
2026-04-02 15:21:10,199 
Hand Evaluation: 7936 samples
2026-04-02 15:21:10,208 
regression: 
         MJE PA_MJE    MVE PAMVE MJE_0 MJE_1 MJE_2  MJE_3  MJE_4  MJE_5  MJE_6  MJE_7  MJE_8  MJE_9 MJE_10 MJE_11 MJE_12 MJE_13 MJE_14 MJE_15 MJE_16 MJE_17 MJE_18 MJE_19 MJE_20
right  13.62   5.75  13.13  5.53  0.00  5.11  8.95  12.21  16.93  10.60  13.99  16.29  20.15  10.95  14.52  16.74  20.73  10.16  13.57  16.24  20.70  10.10  12.78  15.33  19.88
left   14.41   5.57  13.93  5.41  0.00  5.64  9.77  13.19  18.24  11.42  15.04  17.24  21.38  11.82  15.42  17.27  21.57  11.05  14.41  16.76  20.62  11.11  14.04  16.33  20.35
both   14.01   5.66  13.53  5.47  0.00  5.38  9.36  12.70  17.59  11.01  14.51  16.76  20.76  11.38  14.97  17.01  21.15  10.60  13.99  16.50  20.66  10.60  13.41  15.82  20.11
2026-04-02 15:21:10,213 
one_candidate: 
         MJE PA_MJE    MVE PAMVE MJE_0 MJE_1  MJE_2  MJE_3  MJE_4  MJE_5  MJE_6  MJE_7  MJE_8  MJE_9 MJE_10 MJE_11 MJE_12 MJE_13 MJE_14 MJE_15 MJE_16 MJE_17 MJE_18 MJE_19 MJE_20
right  14.30   5.98  13.78  5.74  0.00  5.31   9.34  12.87  17.95  11.10  14.79  17.30  21.31  11.45  15.31  17.70  21.85  10.61  14.24  17.04  21.63  10.51  13.33  16.00  20.68
left   15.22   5.85  14.69  5.67  0.00  5.88  10.25  13.91  19.36  11.98  15.90  18.30  22.59  12.47  16.35  18.42  23.06  11.65  15.24  17.75  21.85  11.64  14.69  17.11  21.27
both   14.76   5.91  14.23  5.71  0.00  5.59   9.80  13.39  18.65  11.54  15.34  17.80  21.95  11.96  15.83  18.05  22.45  11.13  14.74  17.39  21.74  11.07  14.01  16.56  20.97
2026-04-02 15:21:10,218 
agg_candidate: 
         MJE PA_MJE    MVE PAMVE MJE_0 MJE_1 MJE_2  MJE_3  MJE_4  MJE_5  MJE_6  MJE_7  MJE_8  MJE_9 MJE_10 MJE_11 MJE_12 MJE_13 MJE_14 MJE_15 MJE_16 MJE_17 MJE_18 MJE_19 MJE_20
right  12.02   5.64  11.62  5.43  0.00  4.72  8.01  10.63  14.37   9.35  11.95  13.85  17.12   9.64  12.70  14.64  18.31   9.07  12.00  14.45  18.69   9.21  11.58  13.89  18.15
left   12.93   5.49  12.54  5.33  0.00  5.28  8.84  11.54  15.63  10.24  13.24  15.08  18.72  10.67  13.80  15.39  19.29  10.13  13.03  15.12  18.77  10.35  12.95  14.95  18.62
both   12.47   5.57  12.07  5.38  0.00  5.00  8.43  11.09  14.99   9.79  12.59  14.46  17.92  10.15  13.25  15.02  18.80   9.60  12.51  14.78  18.73   9.78  12.26  14.42  18.38
2026-04-02 15:21:10,619 
Object Evaluation: 7936 samples 
Mean Pose: 
                          MCE   MCE2    OCE     ADD   ADDS  ADD01d ADDS01d    REP        CD  FSCORE@2mm  FSCORE@5mm  FSCORE@10mm  FSCORE@2cm  FSCORE@5cm  FSCORE@10cm
002_master_chef_can     77.78  23.05  20.98   53.91  11.66  24.57%  91.00%  29.05  0.011279    0.123535    0.367506     0.631034    0.852364    0.975546     0.997052
003_cracker_box         36.76  29.51  27.15   32.60  14.48  59.31%  91.42%  11.87  0.014277    0.070715    0.288586     0.522651    0.777552    0.964430     0.996175
004_sugar_box           45.65  30.39  25.98   36.67  14.95  41.90%  80.80%  15.37  0.014967    0.113491    0.322879     0.560457    0.793937    0.944596     0.989519
005_tomato_soup_can     49.26  23.90  22.16   37.12  11.94  17.35%  75.90%  17.38  0.011834    0.156969    0.384743     0.629014    0.839524    0.968982     0.996821
006_mustard_bottle      35.26  21.69  19.26   26.90  10.83  60.93%  92.29%  11.45  0.010554    0.144294    0.388225     0.649300    0.869579    0.978258     0.999483
007_tuna_fish_can       57.94  22.23  21.25   39.41  11.85  12.94%  76.90%  19.56  0.011675    0.167119    0.401299     0.654344    0.851279    0.966200     0.993796
008_pudding_box         63.02  47.95  45.65   55.40  32.99  41.20%  79.04%  21.51  0.033091    0.125000    0.327239     0.554331    0.768450    0.917955     0.963961
009_gelatin_box         37.67  27.38  25.59   31.96  14.16  28.72%  76.57%  12.98  0.014234    0.145278    0.360299     0.605953    0.811919    0.942162     0.986910
010_potted_meat_can     45.00  24.30  22.53   34.95  12.50  30.77%  76.92%  16.90  0.012250    0.147694    0.362749     0.610767    0.839083    0.964967     0.995678
011_banana              67.02  37.05  26.98   42.65  19.54  32.43%  66.09%  17.42  0.020462    0.099974    0.254056     0.449345    0.660840    0.889443     0.988103
019_pitcher_base        64.89  40.33  37.92   48.78  20.72  51.13%  88.22%  21.91  0.021192    0.062505    0.257466     0.473369    0.711271    0.912009     0.974201
021_bleach_cleanser     58.93  34.73  28.17   43.02  17.38  49.36%  83.63%  19.62  0.017448    0.088314    0.276116     0.503294    0.739197    0.919473     0.987347
024_bowl               112.25  23.55  21.33   63.58  12.73  18.85%  87.70%  35.82  0.012176    0.133037    0.337045     0.574514    0.815106    0.980650     0.999390
025_mug                 53.00  23.15  22.98   35.27  10.95  25.64%  87.69%  16.10  0.011020    0.149929    0.398046     0.644526    0.852993    0.976408     0.998204
035_power_drill         63.55  37.69  32.93   45.75  19.87  42.60%  77.40%  15.86  0.019342    0.086945    0.261995     0.471979    0.704691    0.902675     0.984360
036_wood_block         145.32  38.08  30.76  102.33  18.58   5.18%  79.84%  59.12  0.019080    0.063861    0.231609     0.439439    0.696322    0.916000     0.985250
037_scissors            70.29  40.59  31.76   43.33  20.89  25.74%  64.46%  17.18  0.021126    0.109943    0.253314     0.416355    0.630104    0.891049     0.985854
040_large_marker        55.62  41.88  36.22   44.77  29.75   7.22%  50.00%  18.82  0.029840    0.127864    0.293790     0.470521    0.678304    0.911292     0.978747
052_extra_large_clamp  121.09  41.81  29.10   62.16  19.97  19.80%  75.63%  30.41  0.019516    0.133116    0.314151     0.481755    0.675263    0.896739     0.987186
061_foam_brick          62.40  25.10  23.07   46.48  12.71  11.52%  69.63%  23.01  0.012592    0.167469    0.409657     0.658860    0.847497    0.958322     0.988035
average_instance        65.68  31.72  27.60   46.09  16.93  30.52%  78.57%  21.39  0.016910    0.120992    0.324799     0.550454    0.771051    0.938932     0.988792
```

Thank you again for your help.

**Comment by zhoujun-7** [comment]
Thanks for your feedback. We did not apply any special settings to produce the results reported in the paper. However, there exist certain discrepancies between the training platforms, which makes it difficult to identify the exact cause of the performance difference. Could you try upgrading your training platform for further validation?
Please let me know if you address this.

**Comment by zhoujun-7** [comment]
We found that the refactoring has somewhat reduced training performance. For more details, please refer to the "Update" section in the README.md. Re-training log file can be found [here](https://github.com/zhoujun-7/VPHO/blob/reproduce/reproduce_training.log). Thanks for pointing this out!
